src/Catalyst.Core.Lib/Util/Duration.cs

```diff
@@ -0,0 +1,201 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+
+namespace Catalyst.Core.Lib.Util
+{
+    /// <summary>
+    ///   Parsing and stringifying of a <see cref="TimeSpan"/> according to IPFS.
+    /// </summary>
+    /// <remarks>
+    ///   From the <see href="https://godoc.org/time#ParseDuration">go spec</see>.
+    ///   <para>
+    ///   A duration string is a possibly signed sequence of decimal numbers, 
+    ///   each with optional fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m". 
+    ///   Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+    ///   </para>
+    /// </remarks>
+    public static class Duration
+    {
+        private const double TicksPerNanosecond = TimeSpan.TicksPerMillisecond * 0.000001;
+        private const double TicksPerMicrosecond = TimeSpan.TicksPerMillisecond * 0.001;
+
+        /// <summary>
+        ///   Converts the string representation of an IPFS duration
+        ///   into its <see cref="TimeSpan"/> equivalent.
+        /// </summary>
+        /// <param name="s">
+        ///   A string that contains the duration to convert.
+        /// </param>
+        /// <returns>
+        ///   A <see cref="TimeSpan"/> that is equivalent to <paramref name="s"/>.
+        /// </returns>
+        /// <exception cref="FormatException">
+        ///   <paramref name="s"/> is not a valid IPFS duration.
+        /// </exception>
+        /// <remarks>
+        ///   An empty string or "n/a" or "unknown" returns <see cref="TimeSpan.Zero"/>.
+        ///   <para>
+        ///   A duration string is a possibly signed sequence of decimal numbers, 
+        ///   each with optional fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m". 
+        ///   Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+        ///   </para>
+        /// </remarks>
+        public static TimeSpan Parse(string s)
+        {
+            if (string.IsNullOrWhiteSpace(s) || s == "n/a" || s == "unknown")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)